examples/user_guide/14-Data_Pipelines.ipynb

```diff
@@ -177,37 +175,18 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "stream = rolling_stream()\n",
-    "\n",
-    "smoothed = rolling(dmap, streams=[stream])\n",
-    "outliers = rolling_outlier_std(dmap, streams=[stream])\n",
+    "outliers = dmap.apply(rolling_outlier_std, rolling_window=slider.param.value)\n",
     "\n",
-    "smoothed * outliers.opts(color='red', marker='triangle')"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "Since the ``rolling_stream`` instance we created is bound to both operations, triggering an event on the stream will trigger both the ``Curve`` and the ``Scatter`` of outliers to be updated:"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "stream.event(rolling_window=50)"
+    "rolled_dmap * outliers.opts(color='red', marker='triangle')"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We can chain operations like this indefinitely and attach streams to each stage. By chaining we can watch our visualization update whenever we change a stream value anywhere in the pipeline and HoloViews will be smart about which parts of the pipeline are recomputed, which allows us to build complex visualizations very quickly.\n",
+    "We can chain operations like this indefinitely and attach parameters or explicit streams to each stage. By chaining we can watch our visualization update whenever we change a stream value anywhere in the pipeline and HoloViews will be smart about which parts of the pipeline are recomputed, which allows us to build complex visualizations very quickly.\n",
     "\n",
-    "In later guides we will discover how to tie custom streams to custom widgets letting us easily control the stream values and making it trivial to define complex dashboards. ``paramNB`` is only one widget framework we could use: we could also choose ``paramBokeh`` to make use of bokeh widgets and deploy the dashboard on bokeh server, or we could manually link ``ipywidgets`` to our streams. For more information on how to deploy bokeh apps from HoloViews and build dashboards see the [Deploying Bokeh Apps](./Deploying_Bokeh_Apps.ipynb) and [Dashboards](./17-Dashboards.ipynb) guides."
+    "In later guides we will see how we can combine HoloViews plots and Panel widgets into custom layouts allowing us to define complex dashboards. For more information on how to deploy bokeh apps from HoloViews and build dashboards see the [Deploying Bokeh Apps](./Deploying_Bokeh_Apps.ipynb) and [Dashboards](./17-Dashboards.ipynb) guides."
```

Comment:
As a general note, I would like to see an `apply` example where the keyword argument is a parameter on an explicitly declared `Stream` object. This doesn't have to be added in this PR (though I think you could make a `RollingWindow` stream example quite easily). One of the implications of `apply` that is worth pointing out is that you can do rebinding of stream parameter names e.g `foo.apply(fn, x=stream_instance.y)`.

Reply:
>As a general note, I would like to see an apply example where the keyword argument is a parameter on an explicitly declared Stream object. This doesn't have to be added in this PR (though I think you could make a RollingWindow stream example quite easily)

I kind of don't see the point. Why wouldn't you just use an ordinary Parameterized? It's more explicit and imo conceptually simpler. Tbh at this point I'd personally never declare any non-linked stream class anymore, whether to reflect that in the documentation is a different question of course.

Reply:
> Tbh at this point I'd personally never declare any non-linked stream class anymore...

I think I can buy that argument in which case the streams documentation would have to talk about the parameter based approach instead, describing linked streams and not mention the explicitly declared custom stream classes.